### PR TITLE
Feature/nuget sources

### PR DESCRIPTION
**Pull request by seesharper**
This PR adds support for specifying package sources when executing a script.

The syntax is identical to the syntax used by the `dotnet cli`

```
dotnet script foo.csx -s http://SomePackageSource
```

It is implemented simply by forwarding the package sources to the `dotnet restore` command we use to resolve runtime and compilation dependencies. 

This change kind of rippled through the code base as we need to pass these package sources down the line. 

Also added some logging to the `CommandRunner`class so that we log the command and the arguments used to execute the command.

### CODE REVIEW

**Comment by filipw** [comment]
ooooh 💄✨

File: src/Dotnet.Script.Core/ScriptContext.cs [hunk 1]
```diff
@@ -9,14 +10,15 @@ namespace Dotnet.Script.Core
 {
     public class ScriptContext
     {
-        public ScriptContext(SourceText code, string workingDirectory, IEnumerable<string> args, string filePath = null, OptimizationLevel optimizationLevel = OptimizationLevel.Debug, ScriptMode scriptMode = ScriptMode.Script)
+        public ScriptContext(SourceText code, string workingDirectory, IEnumerable<string> args, string filePath = null, OptimizationLevel optimizationLevel = OptimizationLevel.Debug, ScriptMode scriptMode = ScriptMode.Script, string[] packageSources = null)
```
Comment (filipw) [inline src/Dotnet.Script.Core/ScriptContext.cs]: these constructors are starting to resemble some of the constructors in Roslyn 😀

**Comment by filipw** [comment]
> It is implemented simply by forwarding the package sources to the dotnet restore command we use to resolve runtime and compilation dependencies.

so not written into csproj? I wonder why we didn't think of that earlier

**Comment by seesharper** [comment]
I actually started to put them into the project file when I suddenly realized that we could just offload this directly to the dotnet cli 😎

**Comment by seesharper** [comment]
I see the build failed. I’ll look into it tomorrow 👍

**Comment by seesharper** [comment]
@filipw Finally green across the board

**Pull request merged**